File: app/src/main/java/ir/mymessage/view/interfaces/MessagesInterface.java
package ir.mymessage.view.interfaces;

import android.content.Context;

import java.util.ArrayList;

import ir.mymessage.model.local.MessageLocal;

public interface MessagesInterface {
    void addMessage(String content, String userId);
    void setupMessagesActivity();
    void displayMessages(ArrayList<MessageLocal> messageArrayList);
    void showSendingStatus();
    void hideSendingStatus();
    void showProgress();
    void hideProgress();
    Context getContext();
}

File: app/src/main/java/ir/mymessage/view/interfaces/MessagesInterface.kt
package ir.mymessage.view.interfaces

import android.content.Context

import java.util.ArrayList

import ir.mymessage.model.local.MessageLocal

interface MessagesInterface {
    fun getContext(): Context
    fun addMessage(content: String, userId: String)
    fun setupMessagesActivity()
    fun displayMessages(messageArrayList: ArrayList<MessageLocal>)
    fun showSendingStatus()
    fun hideSendingStatus()
    fun showProgress()
    fun hideProgress()
}
